File: leetcode/0206/206. Reverse Linked List.py
# Definition for singly-linked list.
class ListNode:
    def __init__(self, val=0, next=None):
        self.val = val
        self.next = next


class Solution:
    def reverseList(self, head: ListNode) -> ListNode:
        pre = None
        cur = head
        while cur:
            next_node = cur.next  # 记录 next
            cur.next = pre  # cur 指向 pre
            pre = cur  # pre 前移
            cur = next_node  # cur 前移
        return pre

File: topic/two_pointers/206. Reverse Linked List.go
package two_pointers

import (
	"algorithm/template"
)

type ListNode = template.ListNode

func reverseList(head *ListNode) *ListNode {
	var pre *ListNode
	var next *ListNode

	for head != nil {
		next = head.Next
		head.Next = pre
		pre = head
		head = next
	}

	return pre
}
